### ISSUE TITLE
Error: Unable to import `../package.json`

### ISSUE THREAD

**Issue by adrhumphreys** [elastic/search-ui#539]
I'm not sure what is causing this, my webpack build chain functions normally, but when you then load the app in the browser you get the error. Perhaps it's an issue with using relative paths and then in dev mode it producing a relative path that the browser is trying to open.

By updating https://github.com/elastic/search-ui/blob/3cfe66660a0b976443122160a4ec28a7b8a1ca5e/packages/search-ui-app-search-connector/src/AppSearchAPIConnector.js#L2 to:
```
const version = '1.5.0';
```

I was able to get around the issue as a very temporary patch

Edit:
Also tried to update it to: `import { version } from "@elastic/search-ui-app-search-connector/package.json";` which still produced the same issue:
```
Uncaught Error: Cannot find module '@elastic/search-ui-app-search-connector/package.json'
    at webpackMissingModule (package.json:1)
    at eval (package.json:1)
    at Object../node_modules/@elastic/search-ui-app-search-connector/package.json (bootstrap:83)
    at __webpack_require__ (bootstrap:19)
    at eval (AppSearchAPIConnector.js:16)
    at Module../node_modules/@elastic/search-ui-app-search-connector/es/AppSearchAPIConnector.js (bootstrap:83)
    at __webpack_require__ (bootstrap:19)
    at eval (index.js:2)
    at Module../node_modules/@elastic/search-ui-app-search-connector/es/index.js (bootstrap:83)
    at __webpack_require__ (bootstrap:19)
```

Edit 2:
Looks like the fool is me, babel was trying to read it and, then not being able too, so it would just exclude it from the build.
Updating my babel config to ignore `*/**/package.json` fixed the issue for me

**Comment by simofacc** [comment]
Hi,

This issue is probably related to webpack 5 named exports https://webpack.js.org/migrate/5/#using-named-exports-from-json-modules.

**Comment by ctestamarck** [comment]
NextJS with Webpack5 shows this warning: `Should not import the named export 'version' (imported as 'version') from default-exporting module (only default export is available soon)`

Was able to patch per Webpack documentation: 

`diff --git a/node_modules/@elastic/search-ui-app-search-connector/es/AppSearchAPIConnector.js b/node_modules/@elastic/search-ui-app-search-connector/es/AppSearchAPIConnector.js
index d6c5e74..cd11fe2 100644
--- a/node_modules/@elastic/search-ui-app-search-connector/es/AppSearchAPIConnector.js
+++ b/node_modules/@elastic/search-ui-app-search-connector/es/AppSearchAPIConnector.js
@@ -10,7 +10,10 @@ function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (O
 function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }
 
 import * as ElasticAppSearch from "@elastic/app-search-javascript";
-import { version } from "../package.json";
+import pkg from '../package.json';
+const version = pkg.version;
+
+
 import { adaptResponse } from "./responseAdapter";
 import { adaptRequest } from "./requestAdapters";
 import buildResponseAdapterOptions from "./buildResponseAdapterOptions"; // The API will error out if empty facets or filters objects
`

**Comment by Pearce-Ropion** [comment]
Why was this closed @adrhumphreys? This is still a bug that affects all users that use this monorepo with webpack 5.

As per @simofacc suggestion, all instances of importing individual keys from the package.json should be updated to import the default export. 

```js
import { version } from "./package.json";
const foo = version; // example
// should be changed to 
import packageJson from "./package.json";
const foo = packageJson.version; // example
```

There are at least 3 instances of this throughout the repo
https://github.com/elastic/search-ui/search?p=1&q=import+%7B+version+%7D

Without the change this warning occurs and will eventually break builds once the feature is permanently removed from webpack.
![image](https://user-images.githubusercontent.com/32177944/123727952-5aa56b00-d847-11eb-80a0-e264ad76db2d.png)

**Comment by JasonStoltz** [comment]
Could someone please verify this fix in 1.7?

**Comment by Pearce-Ropion** [comment]
I have verified that this is fixed in 1.7.
